scalafix-core/shared/src/main/scala/scalafix/internal/config/DisableSyntaxConfig.scala

```diff
@@ -0,0 +1,191 @@
+package scalafix.internal.config
+
+import metaconfig.{Conf, ConfError, ConfDecoder, Configured}
+import org.langmeta._
+import MetaconfigPendingUpstream.XtensionConfScalafix
+
+import scala.meta.tokens.Token
+
+case class DisableSyntaxConfig(
+    keywords: Set[DisabledKeyword] = Set(),
+    tab: Boolean = false,
+    semicolon: Boolean = false,
+    xml: Boolean = false
+) {
+  implicit val reader: ConfDecoder[DisableSyntaxConfig] =
+    ConfDecoder.instanceF[DisableSyntaxConfig](c =>
+      (
+        c.getField(keywords) |@|
+          c.getField(tab) |@|
+          c.getField(semicolon) |@|
+          c.getField(xml)
+      ).map {
+        case (((a, b), c), d) =>
+          DisableSyntaxConfig(a, b, c, d)
+      }
+    )
+
+  def isDisabled(keyword: String): Boolean =
+    keywords.contains(DisabledKeyword(keyword))
+}
+
+object DisableSyntaxConfig {
+  val default = DisableSyntaxConfig()
+  implicit val reader: ConfDecoder[DisableSyntaxConfig] = default.reader
+}
+
+case class DisabledKeyword(keyword: String)
+
+object DisabledKeyword {
+  def unapply(token: Token): Option[String] = {
+    import Token._
+
+    token match {
```

Comment:
Can we avoid this match and use product prefix instead? This approach is unsafer since there's a risk we accidentally add a typo in the match. We should also avoid doing a large match like this on every token.

Reply:
The config reader already validates that the configuration contains only valid keywords.